lib/pos_mempool.go

```diff
@@ -60,6 +60,60 @@ type Mempool interface {
 	) (uint64, error)
 }
 
+// GetAugmentedUniversalViewWithAdditionalTransactions is meant as a helper function
+// for backend APIs to better construct atomic transactions while maintaining various
+// transactional sanity checks. It SHOULD NOT be used for consensus critical tasks
+// as it does not validate signatures, does not validate fees, and would likely lead
+// to a nonsensical state of the blockchain.
+//
+// In the case of atomic transactions it's likely that a user will have a series of
+// dependent transactions (transactions that MUST be submitted together in a specific order)
+// that they plan to submit as a single atomic transaction. However, functions like
+// GetAugmentedUniversalView may not have access to this series of transactions meaning
+// backend APIs using GetAugmentedUniversalView will generate errors unnecessarily in the case
+// of certain atomic transaction workflows. To deal with this, we can use
+// GetAugmentedUniversalViewWithAdditionalTransactions which will create a
+// view that has connected a set of transactions (specified by optionalTxns).
+//
+// NOTE: GetAugmentedUniversalViewWithAdditionalTransactions DOES NOT validate fees
+// as fees are computed in UtxoView.ConnectBlock and optionalTxns are not included
+// in any block that can be connected yet.
+func GetAugmentedUniversalViewWithAdditionalTransactions(
+	mempool Mempool,
+	optionalTxnsBlockHeight uint32,
+	optionalTxnsTimestampNanoSecs int64,
```

Comment:
Any reason you want the block height and timestamp to be parametrizable? Ideally, these are decided by the mempool based on the chain current height + 1 and the current timestamp. That way, a caller can't connect the txn in a way that doesn't reflect how the txns will be connect on-chain

Reply:
I addressed this concern by adding in a "GetMempoolTipBlockHeight() uint64" method to the Mempool interface and implementing it for both PoS Mempool and legacy mempool. Now you only must pass the mempool and optionalTxns to generate a UtxoView. 

I opted to make the interface method get the tip block height as this seemed the most universally useful function moving forward.